aws/acm.go

```diff
@@ -1,227 +1,69 @@
 package aws
 
 import (
-	"log"
-	"strings"
 	"sync"
-	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
 	"github.com/aws/aws-sdk-go/service/acm"
 	"github.com/aws/aws-sdk-go/service/acm/acmiface"
 )
 
-// CertDetail is the business object for Certificates
-type CertDetail struct {
-	Arn       string
-	AltNames  []string
-	NotBefore time.Time
-	NotAfter  time.Time
+type acmClient struct {
```

Comment:
Not exactly. You only need a specify constructor function for each individual CertificatesProvider.

Something like `cache := newCache(newIAMCertProvider(iamAPI), newACMCertProvider(acmAPI))`

From there own, it's the common interface sig.